src/components/playback/playbackmanager.js

```diff
@@ -1922,58 +1908,115 @@ class PlaybackManager {
-                            resolve(null);
-                            return;
+            return null;
+        }
+
+        async function getSeriesOrSeasonPlaybackPromise(firstItem) {
+            const apiClient = ServerConnections.getApiClient(firstItem.ServerId);
+
+            const episodesResult = await apiClient.getEpisodes(firstItem.SeriesId || firstItem.Id, {
+                IsVirtualUnaired: false,
+                IsMissing: false,
+                UserId: apiClient.getCurrentUserId(),
+                Fields: ['Chapters', 'Trickplay']
+            });
+            const originalResults = episodesResult.Items;
+            const isSeries = firstItem.Type === 'Series';
+            let foundItem = false;
+
+            episodesResult.Items = episodesResult.Items.filter(function (e) {
+                if (foundItem) {
+                    return true;
+                }
+
+                if (!e.UserData.Played && (isSeries || e.SeasonId === firstItem.Id)) {
+                    foundItem = true;
+                    return true;
+                }
+
+                return false;
+            });
+
+            if (episodesResult.Items.length === 0) {
+                if (isSeries) {
+                    episodesResult.Items = originalResults;
+                } else {
+                    episodesResult.Items = originalResults.filter(function (e) {
+                        if (foundItem) {
+                            return true;
                         }
 
-                        apiClient.getEpisodes(firstItem.SeriesId, {
-                            IsVirtualUnaired: false,
-                            IsMissing: false,
-                            UserId: apiClient.getCurrentUserId(),
-                            Fields: ['Chapters', 'Trickplay']
-                        }).then(function (episodesResult) {
-                            let foundItem = false;
-                            episodesResult.Items = episodesResult.Items.filter(function (e) {
-                                if (foundItem) {
-                                    return true;
-                                }
-                                if (e.Id === firstItem.Id) {
-                                    foundItem = true;
-                                    return true;
-                                }
+                        if (e.SeasonId === firstItem.Id) {
+                            foundItem = true;
+                            return true;
+                        }
 
-                                return false;
-                            });
-                            episodesResult.TotalRecordCount = episodesResult.Items.length;
-                            resolve(episodesResult);
-                        }, reject);
+                        return false;
                     });
-                });
+                }
             }
```

Comment:
These changes should be reverted because they revert https://github.com/jellyfin/jellyfin-web/pull/5415
_Merge conflicts were incorrectly resolved._

Reply:
I updated the refactor to include the new changes. I believe it brings it back in parity with #5415